providers/userdetail/src/java/org/sakaiproject/user/detail/ValueEncryptionService.java

```diff
@@ -0,0 +1,119 @@
+package org.sakaiproject.user.detail;
+
+import java.security.AlgorithmParameters;
+import java.security.SecureRandom;
+import java.security.spec.KeySpec;
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.sakaiproject.component.cover.ServerConfigurationService;
+
+/**
+ * This class provides encryption/decryption services. Service is thread safe.
+ */
+public class ValueEncryptionService {
```

Comment:
You can still have it as a service, but not exposed in the API. The only part that stops it being easily testable is the call to ServerConfigurationService, so you could pass the key in.

Or just rename to make it clearer it's not a service.